packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -11,42 +11,164 @@
+import styled from "@emotion/styled";
 
-import { useOdysseyDesignTokens } from "../OdysseyDesignTokensContext";
 import { Box } from "../Box";
 import { FieldComponentProps } from "../FieldComponentProps";
 import { FieldHint } from "../FieldHint";
+import { CheckedFieldProps } from "../FormCheckedProps";
 import type { HtmlProps } from "../HtmlProps";
-import { useUniqueId } from "../useUniqueId";
+import { CheckIcon } from "../icons.generated";
 import { ComponentControlledState, getControlState } from "../inputUtils";
-import { CheckedFieldProps } from "../FormCheckedProps";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+import { useUniqueId } from "../useUniqueId";
 
 const { CONTROLLED } = ComponentControlledState;
 
+const stripRemFromUnits = (remValue: string) => Number(remValue.split("r")[0]);
+
+const SwitchAndLabelContainer = styled.div<
+  Pick<SwitchProps, "isFullWidth" | "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isFullWidth, odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "flex-start",
+  justifyContent: "space-between",
+  maxWidth: isFullWidth ? "100%" : odysseyDesignTokens.TypographyLineLengthMax,
+}));
+
+const SwitchContainer = styled.div({
+  position: "relative",
+});
+
+const StyledSwitchLabel = styled(FormLabel, {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<
+  Pick<SwitchLabelComponentProps, "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isDisabled, odysseyDesignTokens }) => ({
+  display: "block",
+  margin: 0,
+  color: isDisabled
+    ? odysseyDesignTokens.TypographyColorDisabled
+    : odysseyDesignTokens.PaletteNeutralDark,
+
+  ...(isDisabled && {
+    p: {
+      color: odysseyDesignTokens.TypographyColorDisabled,
+    },
+    a: {
+      color: `${odysseyDesignTokens.TypographyColorDisabled} !important`,
+    },
+  }),
+}));
+
+const SwitchTrack = styled.div<
+  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isChecked, isDisabled, odysseyDesignTokens }) => ({
+  position: "relative",
+  width: odysseyDesignTokens.Spacing7,
+  height: `calc(${odysseyDesignTokens.Spacing4} + ${odysseyDesignTokens.Spacing1})`,
+  borderRadius: odysseyDesignTokens.BorderRadiusOuter,
+  backgroundColor: isDisabled
+    ? odysseyDesignTokens.HueNeutral200
+    : isChecked
+      ? odysseyDesignTokens.PaletteSuccessLight
+      : odysseyDesignTokens.HueNeutral300,
+  transition: `background-color ${odysseyDesignTokens.TransitionDurationMain}`,
+}));
+
+const SwitchThumb = styled.span<
+  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isChecked, isDisabled, odysseyDesignTokens }) => {
+  const thumbOffset = (3 / 16) * (16 / 14);
+  const trackWidth = stripRemFromUnits(odysseyDesignTokens.Spacing7);
+  const thumbWidth =
+    stripRemFromUnits(odysseyDesignTokens.Spacing4) - (2 / 16) * (16 / 14);
```

Comment:
Can we get a comment here to describe what this math is doing?

2/16 * 16/14? <- I get that the 16/14 is swapping into 16 pixel mode rather than our 14 pixel mode, but the way it's written, I'm not sure I understand why it's after the 2/16 math.

Can we convert it, then subtract our the pixels?

What's 2/16 supposed to mean here anyway? Seems arbitrary.

Reply:
The idea is to convert 2px into a rem value